src/modules/rlm_eap/libeap/eap_tls.c

```diff
@@ -160,15 +164,36 @@ int eaptls_success(eap_handler_t *handler, int peap_flag)
 	/*
 	 *	Automatically generate MPPE keying material.
 	 */
-	if (tls_session->prf_label) {
+	if (tls_session->label) {
+		switch (tls_session->info.version) {
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+		case TLS1_3_VERSION:
+			context = context_tls13;
+			context_size = context_tls13_size;
+			break;
+#endif
+		case TLS1_2_VERSION:
+		case TLS1_1_VERSION:
+		case TLS1_VERSION:
+			break;
+		case SSL2_VERSION:
+		case SSL3_VERSION:
+		default:
+			/* Should never happen */
+			rad_assert(0);
+			return 0;
+			break;
+		}
 		eaptls_gen_mppe_keys(handler->request,
-				     tls_session->ssl, tls_session->prf_label);
+				     tls_session->ssl, tls_session->label,
+				     context, context_size);
 	} else if (handler->type != PW_EAP_FAST) {
 		RWDEBUG("Not adding MPPE keys because there is no PRF label");
 	}
 
-	eaptls_gen_eap_key(handler->request->reply, tls_session->ssl,
-			   handler->type);
+	rad_assert(context_tls13[0] == handler->type);	/* expected by eaptls_gen_eap_key */
+	eaptls_gen_eap_key(handler->request->reply, tls_session->ssl, context_tls13, context_tls13_size);
```

Comment:
shouldn't this be `context` and `context_size`, instead of hard-coded tls13 contexts?

Reply:
Resolved by https://github.com/FreeRADIUS/freeradius-server/pull/3516/commits/eced45c8153bc0d091450e4533ec49dddcddc86d